### PR TITLE
[KYUUBI #5374] JDBC Engine supports ClickHouse

### LINKED ISSUE #5374: [TASK][MEDIUM] JDBC Engine supports ClickHouse
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/296) and found no similar tasks.

### Mentor

- [ ] I have sufficient knowledge and experience of this task, and I volunteer to be the mentor of this task to guide contributors to complete the task.

### Skill requirements

- Basic knowledge of Scala programming language
- Basic knowledge of JDBC concepts
- Basic knowledge of Docker and [Testcontianers](https://java.testcontainers.org/)

### Background and Goals

Currently, Kyuubi supports JDBC engines with limited dialects, e.g. Doris, Phoenix, we can easily extend the dialects to support more databases.

Reference https://www.bilibili.com/video/BV1UT411m7LR

### Implementation steps

- Implement a ClickHouse dialect in the JDBC engine
- Leverage the Testcontainers to add integration tests
- Update document correspondingly

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lsm1**
# :mag: Description
## Issue References 🔗

This pull request fixes #5374 

## Describe Your Solution 🔧

JDBC Engine supports ClickHouse

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: integration-tests/kyuubi-jdbc-it/pom.xml [hunk 1]
```diff
@@ -96,6 +96,24 @@
...
+
+        <dependency>
+            <groupId>com.dimafeng</groupId>
+            <artifactId>testcontainers-scala-clickhouse_${scala.binary.version}</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.clickhouse</groupId>
+            <artifactId>clickhouse-jdbc</artifactId>
```
Comment (pan3793) [inline integration-tests/kyuubi-jdbc-it/pom.xml:108]: can we use the `http` classifier jar? then we don't need to care about lz4 jar

Comment (lsm1) [reply integration-tests/kyuubi-jdbc-it/pom.xml:108]: ok,fix it

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -3067,7 +3067,8 @@ object KyuubiConf {
         "<li>mysql: For establishing MySQL connections.</li> " +
         "<li>phoenix: For establishing Phoenix connections.</li> " +
         "<li>postgresql: For establishing PostgreSQL connections.</li>" +
-        "<li>starrocks: For establishing StarRocks connections.</li>")
+        "<li>starrocks: For establishing StarRocks connections.</li>" +
+        "<li>clickhouse: For establishing clickhouse connections.</li>")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3071]: seems we forgot to add Impala, could u please open a new PR to add Impala case?

File: pom.xml [hunk 3]
```diff
@@ -190,6 +190,7 @@
         <scalatestplus.version>3.2.16.0</scalatestplus.version>
         <scopt.version>4.1.0</scopt.version>
         <slf4j.version>1.7.36</slf4j.version>
+        <ck.jdbc.version>0.6.0</ck.jdbc.version>
```
Comment (pan3793) [inline pom.xml:193]:
```suggestion
        <clickhouse-java.version>0.6.0</ck.jdbc.version>
```

**Review by pan3793 — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 2 above)
Comment (lsm1) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3071]: ok

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**